### PR TITLE
Add livephotos support

### PR DESCRIPTION
**Pull request by fefux**
Add livephotos support from Apple :
- Create a new container in slideshow view
- Authorize loading data from "blob:" in content security policy
- Retrieve NPM version of livephotoskit.js

To work, a .jpg/.JPG picture needs to have beside a .mov/.MOV video with the same name

Signed-off-by: François Sylvestre <francoissylv@gmail.com>

### CODE REVIEW

**Comment by jancborchardt** [comment]
Thanks a lot for your contribution @fefux! :tada: And sorry for the late reply.

Could someone with macOS or iPhone look into this? @nextcloud/mac :green_apple:

**Comment by jospoortvliet** [comment]
@fefux thanks, esp given we're close to the 14 release :1st_place_medal:  - I think we should try to get this in if we still can... If not, 15 :2nd_place_medal:

**Comment by fefux** [comment]
Just before going in vacation, I tested a functional patch. The issue is the default slideshow is override by the one in gallery but additional dependencies for livephoto not.
I don't have my laptop so I can't do the pullrequest before the 20th.
@jospoortvliet  I saw release date for release 14 is the 29th. I will do my possible but it seems quite close to do review and tests before merging in time.

**Comment by MorrisJobke** [comment]
@fefux Even if this is a super nice feature, time runs up and we need to move this to Nextcloud 15, which is planned for December. I hope this is okay for you?

**Comment by fefux** [comment]
No problem for me, in addition this functionnality is eventually more usefull if nextcloud app for iphone supports importation of live photo too (which seems not be the case now).

**Comment by AndreiButoi** [comment]
I saw that ios live photos support was removed from the Nextcloud 15 milestone. Will still be developed?

**Comment by fefux** [comment]
Hi everybody, sorry for the late, I just updated the code for support livephoto for nextcloud 15.0.4 and with the tip of the branch. @AndreiButoi development still on going.
I see that my pullrequest have many problems with CI tools. Is my code is in fault or is tools/tests need to be updated (for example, I saw an error with signed-off for CI/drone, but my commit contains a signed off)? Have a nice day

File: appinfo/signature.json [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+}
```
Comment (MorrisJobke) [inline appinfo/signature.json]: As this changes once a file is touched it should not be checked into git but only be generated during packaging the app for the app store release.

Comment (fefux) [reply appinfo/signature.json]: Thanks for the info, the file has been removed from repository

**Comment by SimonSelg** [comment]
@fefux do you need some help on this? I'd be willing to work on this issue as I need it myself :)

**Comment by MorrisJobke** [comment]
Joe do you extract those two files from an iPhone? We tried to extract this, but it always just exported the jpeg file without the mov.

Also @skjnldsv is working on bringing this to the viewer (see repo at nextcloud/viewer repo)

**Comment by maci2100** [comment]
Something new here ?

**Comment by skjnldsv** [comment]
> Something new here ?

This is already in the new viewer :)

**Comment by MorrisJobke** [comment]
https://github.com/nextcloud/viewer <- this unifies the different viewers of gallery, text editor, PDF, etc and replaces the slideshow from the gallery app.

**Comment by k1l1** [comment]
Is it already released in NC 16. I doesn't work for me. When I use the viewer in a Folder which contains **IMG_4112.JPG** and **IMG_4112.MOV** one is shown as picture, the other as video. What is the expected behaviour?

**Comment by skjnldsv** [comment]
@k1l1 nc 17

**Comment by thibaultamartin** [comment]
@skjnldsv will this actually be in Nextcloud 17? It looks like it will wait for NC18 at the very least, doesn’t it?

**Comment by skjnldsv** [comment]
> @skjnldsv will this actually be in Nextcloud 17? It looks like it will wait for NC18 at the very least, doesn’t it?

We now ship a new viewer since 16. Not the gallery, on files. It now sypport livephotos that are split into jpg/video in the same dir :)

**Comment by thibaultamartin** [comment]
> We now ship a new viewer since 16. Not the gallery, on files.

Okay so just to make sure I'm getting it right: let's consider I upload a Live Photo from my iPhone automatically using Nextcloud's app. A new folder with the photo + video of the Live Photo is created in my /Photo folder. If I browse to said folder using my instance's file explorer, the file viewer will display it as a Live Photo. If I browse my Gallery, I will see a photo and a video. Is this right?

> It now sypport livephotos that are split into jpg/video in the same dir :)

It may not belong here, but still has some ties with the gallery I think: if I upload a Live Photo from my phone to the Gallery, wipe the phone, and download back what was uploaded to my instance, will I have a Live Photo or a photo+video on my phone?

Thanks for your answers and work @skjnldsv, it's really helpful :)

**Comment by skjnldsv** [comment]
> will I have a Live Photo or a photo+video on my phone

I cannot tell. This is something to check with your phone :)

> If I browse my Gallery, I will see a photo and a video. Is this right?

Right! :) 
The gallery is a bit old and unusable as is. We currently have the grid view which is similar

**Comment by thibaultamartin** [comment]
Crystal clear, thanks! All this leads me to think that gallery is going to be either deprecated or deeply changed. Is this the case?

**Comment by skjnldsv** [comment]
> Crystal clear, thanks! All this leads me to think that gallery is going to be either deprecated or deeply changed. Is this the case?

Yes :)

**Comment by adutta98** [comment]
> > Crystal clear, thanks! All this leads me to think that gallery is going to be either deprecated or deeply changed. Is this the case?
> 
> Yes :)
So if I understand this right, the requirement for this to work is a jpg/mov combo, is there any chance it would also work with a HEIC/mov combo, files of the same name? or any wip for it?

Thanks for all the good work.

**Comment by skjnldsv** [comment]
> jpg/mov combo

Not only, any image/video combo is fine :)

**Comment by skjnldsv** [comment]
> would also work with a HEIC/mov combo

We just need to support heic then, but that's a tricky part since heic can contains lots of stuff: multiple images, videos...

**Comment by logo000** [comment]
Hello, 
First, I love your work. Nextcloud is a awesome program. Iam a iOS User and my love are Live Photos. This is the only Feature that not implemented yet. So is there any progress or eta for this feature? Thx. So much <3

**Comment by nickvergessen** [comment]
The gallery app has been replaced by the beautiful new app:
[Nextcloud Photos - :camera_flash: Your memories under your control](https://github.com/nextcloud/photos/)

Please checkout if your Pull request is still necessary there, and in case create it there or raise an issue for others to copy the change from here.

**Pull request closed without merging**